Add comprehensive holiday categories documentation

tharu-jwd opened the pull request:
## Summary

This PR addresses issue #1717 by adding comprehensive documentation for holiday categories.

### Changes Made

- **Created `docs/holiday_categories.md`**: Complete documentation covering all available holiday categories including:
  - Official status categories (PUBLIC, GOVERNMENT, WORKDAY, UNOFFICIAL)
  - Institutional categories (BANK, SCHOOL, ARMED_FORCES)  
  - Religious categories (CATHOLIC, ISLAMIC, HINDU, ORTHODOX, etc.)
  - Ethnic/cultural categories (ARMENIAN, TURKISH, SERBIAN, etc.)
  - Special categories (OPTIONAL, MANDATORY, HALF_DAY)

- **Enhanced `docs/examples.md`**: Improved the holiday categories section with practical usage examples and reference link to detailed documentation

- **Added contributor**: Added name to CONTRIBUTORS file

### Documentation Features

- Comprehensive category definitions with explanations
- Practical code examples for single and multiple category usage
- Implementation guidelines for contributors
- Country-specific usage examples

### Test Coverage

This is a documentation-only change. All examples have been tested to ensure they work correctly.

Closes #1717

tharu-jwd committed:
Add comprehensive holiday categories documentation

- Create detailed holiday_categories.md documentation
- Enhance examples.md with improved category usage examples
- Add professional documentation for all 25+ category types
- Include usage examples and implementation guidelines
- Add contributor name to CONTRIBUTORS file

Addresses issue #1717

tharu-jwd committed:
Fix category format consistency in documentation examples

Address Copilot feedback:
- Use constants format (BANK, PUBLIC) instead of strings ('bank', 'public')
- Add proper imports for holidays.constants
- Ensure consistent format across all examples
- All examples tested and verified to work correctly

tharu-jwd commented on docs/holiday_categories.md:
```diff
@@ -0,0 +1,288 @@
+import holidays
+
+# Check which categories a country supports
+us = holidays.UnitedStates()
```
resolved

tharu-jwd commented on docs/examples.md:
````diff
@@ -143,16 +143,35 @@ To change the language translation, you can set the language explicitly.
+
+``` python
+>>> # Get both public and bank holidays for Belgium
+>>> for dt, name in sorted(holidays.BE(years=2023, language="en_US", categories=('bank', 'public')).items()):
````
done

tharu-jwd committed:
Address CodeRabbit feedback and improve documentation clarity

- Add concrete examples for SABIAN and YAZIDI categories
- Clarify UNOFFICIAL category criteria (exclude novelty/marketing days)
- Add examples for all ethnic/cultural categories (ARMENIAN, ALBANIAN, etc.)
- Add stability warning about category tuples changing over time
- Add note about case-sensitive category matching
- Improve overall documentation completeness per review feedback

tharu-jwd committed:
Address KJhellico feedback on holiday categories documentation

- Replace string category values with constants from holidays.constants for better maintainability
- Fix Germany Catholic holidays example to specify Saxony subdivision (SN) since Catholic holidays only apply to specific German states
- Update ethnic community examples to consistently reference North Macedonia:
  - Bosnian category: Changed from Bosnia and Herzegovina to North Macedonia
  - Serbian category: Changed from Montenegro and Bosnia to North Macedonia
  - Roma category: Changed from various European countries to North Macedonia
  - Vlach category: Changed from Romania and other Balkan countries to North Macedonia

PPsyrius commented on docs/examples.md:
```diff
@@ -143,15 +143,38 @@ To change the language translation, you can set the language explicitly.
-2023-12-26 Bank Holiday
+>>> import holidays
+>>> from holidays.constants import PUBLIC, UNOFFICIAL
+>>> # Get only public holidays
```
Code comments like this should end with a dot (`.`) - same goes for other entries 👍 
```suggestion
>>> # Get only public holidays.
```

PPsyrius commented on docs/examples.md:
````diff
@@ -171,6 +194,39 @@ To get a list of other categories holidays (for countries that support them):
+``` python
+>>> from holidays.constants import CATHOLIC
+>>> # Get Catholic holidays in Germany (Saxony subdivision)
+>>> de_catholic = holidays.Germany(subdiv='SN', categories=CATHOLIC, years=2024)
+>>> for date, name in sorted(de_catholic.items()):
````
Just my 2-cents, but let's wait for KJ's input on this particular one
```suggestion
>>> de_sn_catholic = holidays.Germany(subdiv='SN', categories=CATHOLIC, years=2024)
>>> for date, name in sorted(de_sn_catholic.items()):
```

KJhellico replied:
Let's be `de_sn_catholic`, for accuracy.

tharu-jwd replied:
Done

PPsyrius commented on docs/holiday_categories.md:
```diff
@@ -0,0 +1,322 @@
+
+### Official Status Categories
+
+**PUBLIC**
```
Maybe add another newline here or switch to `####` as this is how it currently looks like when rendered
```suggestion
**PUBLIC**

```
```suggestion
#### PUBLIC

```
<img width="1052" height="652" alt="image" src="https://github.com/user-attachments/assets/2427de99-c732-403c-b076-fd58a9bb097d" />

tharu-jwd replied:
changed

PPsyrius commented on docs/holiday_categories.md:
```diff
@@ -0,0 +1,322 @@
+- Cultural observances with optional recognition
+
+**MANDATORY**
+Holidays that must be observed by law in specific contexts.
```
AFAIK `MANDATORY` is currently only used by Macau at the moment for its Statutory Holidays that must be day-off i.e. https://www.dsal.gov.mo/en/standard/holiday_table.html (its public list currently represents its "General" Holidays list.

@KJhellico @arkid15r Do we want to phase out this particular category and move the "General" list to `OPTIONAL` category instead, like Hong Kong's implementation?

KJhellico replied:
Yes, I think it will be better. MANDATORY name doesn't look intuitive. :)

tharu-jwd committed:
Address PPsyrius feedback on documentation formatting

- Add periods to all code comments for consistency
- Fix PUBLIC section formatting with proper spacing

tharu-jwd committed:
Add deprecation notice to MANDATORY category

Based on maintainer discussion, MANDATORY category is planned for deprecation
in favor of OPTIONAL category to improve clarity and consistency.

tharu-jwd committed:
Add blank lines after Examples headings

tharu-jwd commented on docs/holiday_categories.md:
```diff
@@ -0,0 +1,323 @@
+- New Year's Day in most countries
+- Independence Day in the United States
+- Christmas Day in Christian-majority countries
+
```
Done. please check.

tharu-jwd committed:
Convert category headings to #### format for better hierarchy

PPsyrius commented on docs/holiday_categories.md:
````diff
@@ -0,0 +1,373 @@
+# Note: The category tuples shown are examples and may change as countries update their holiday frameworks.
+```
+
+### Examples by Country
+
+**United States**: `('government', 'public', 'unofficial')`
+- Supports federal holidays, state holidays, and cultural observances
+
+**Germany**: `('catholic', 'public')`
+- Supports public holidays and Catholic religious holidays in certain regions
+
+**India**: `('optional', 'public')`
+- Supports national holidays and optional regional observances
+
+**Argentina**: `('armenian', 'bank', 'government', 'hebrew', 'islamic', 'public')`
+- Supports multiple religious traditions and institutional categories
+
````
The current implementation looks fine as-is when rendered
<img width="1042" height="728" alt="image" src="https://github.com/user-attachments/assets/1c751ebb-e6a1-4483-ad10-2ed3460fb578" />

PPsyrius reviewed (commented):
Aside from the pending comments, the documentation is in good shape now IMO 👍

tharu-jwd committed:
Update variable name to de_sn_catholic for clarity

Changed de_catholic to de_sn_catholic in the Germany Saxony example
to better reflect the specific subdivision being used.

tharu-jwd committed:
Update MANDATORY category to deprecated status

Reflect that MANDATORY category is now deprecated rather than planned
for deprecation, with updated description of Macau reclassification.

tharu-jwd commented on docs/holiday_categories.md:
```diff
@@ -0,0 +1,373 @@
+
+#### MANDATORY *(Planned for deprecation)*
+
+Holidays that must be observed by law in specific contexts. Currently only used by Macau. This category may be phased out in favor of OPTIONAL to improve clarity and consistency.
```
@PPsyrius @KJhellico so we keep MANDATORY as it is?

KJhellico replied:
Let's keep it now.

arkid15r committed:
Merge branch 'dev' into pr/tharu-jwd/2851

arkid15r committed:
Update docs

arkid15r committed:
Merge branch 'dev' into document-holiday-categories

KJhellico committed:
Fix format

arkid15r reviewed (approved):
@tharu-jwd thanks for adding this.

Sorry it took that long to merge -- totally my fault.

KJhellico reviewed (approved):
LGTM.
